src/Magento/FunctionalTestingFramework/Extension/BrowserLogUtil.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\FunctionalTestingFramework\Extension;
+
+/**
+ * Class BrowserLogUtil
+ * @package Magento\FunctionalTestingFramework\Extension
+ */
+class BrowserLogUtil
+{
+    const LOG_TYPE_BROWSER = "browser";
+    const ERROR_TYPE_JAVASCRIPT = "javascript";
+
+    /**
+     * Loops through stepEvent for browser log entries
+     *
+     * @param \Magento\FunctionalTestingFramework\Module\MagentoWebDriver $module
+     * @param \Codeception\Event\StepEvent                                $stepEvent
+     * @return void
+     */
+    public static function logErrors($module, $stepEvent)
+    {
+        //Types available should be "server", "browser", "driver". Only care about browser at the moment.
+        if (in_array(self::LOG_TYPE_BROWSER, $module->webDriver->manage()->getAvailableLogTypes())) {
+            $browserLogEntries = $module->webDriver->manage()->getLog(self::LOG_TYPE_BROWSER);
+            $jsErrors = self::getLogsOfType($browserLogEntries, self::ERROR_TYPE_JAVASCRIPT);
+            foreach ($jsErrors as $entry) {
+                self::logError(self::ERROR_TYPE_JAVASCRIPT, $stepEvent, $entry);
+                //Set javascript error in MagentoWebDriver internal array
+                $module->setJsError("ERROR({$entry["level"]}) - " . $entry["message"]);
+            }
+        }
+    }
+
+    /**
+     * Loops through given log and returns entries of the given type.
+     *
+     * @param array  $log
+     * @param string $type
+     * @return array
+     */
+    public static function getLogsOfType($log, $type)
+    {
+        $errors = [];
+        foreach ($log as $entry) {
+            if (array_key_exists("source", $entry) && $entry["source"] === $type) {
+                $errors[] = $entry;
+            }
+        }
+        return $errors;
+    }
+
+    /**
+     * Loops through given log and filters entries of the given type.
+     *
+     * @param array  $log
+     * @param string $type
+     * @return array
+     */
+    public static function filterLogsOfType($log, $type)
```

Comment:
`filterLogsOfType()` and `getLogsOfType` are complete the same. Any reason we need two functions here?

Reply:
They are almost completely the same, check ln50 vs ln68

Reply:
Ok, I see the difference.